perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ResourcesManagerBlImpl.java

```diff
@@ -1175,6 +1181,19 @@ public void deactivateGroupResourceAssignment(PerunSession sess, Group group, Re
 
 		getResourcesManagerImpl().setGroupResourceStatus(sess, group, resource, GroupResourceStatus.INACTIVE);
 		getPerunBl().getAuditer().log(sess, new GroupRemovedFromResource(group, resource));
+
+		// reset assignment failure cause
+		getResourcesManagerImpl().setFailedGroupResourceAssignmentCause(sess, group, resource, null);
+	}
+
+	@Override
+	public String getFailedGroupResourceAssignmentCause(PerunSession sess, Group group, Resource resource) throws GroupNotDefinedOnResourceException, GroupResourceStatusException {
+		GroupResourceStatus status = getResourcesManagerImpl().getGroupResourceStatus(sess, group, resource);
+		if (status != GroupResourceStatus.FAILED) {
+			throw new GroupResourceStatusException("Cannot read assignment failure cause for not FAILED assignment.");
+		}
```

Comment:
In such scenarios, it is better to just return `null` value (or `Optional<String>` if you want to do it the best way). If you throw an exception, you limit the usage of such a method (you force the method users to handle the exception even if they first check that the status is failing for example).

Reply:
Added `failureCause` to `AssignedResource`, `AssignedGroup` and `GroupResourceAssignment`, replaced throwing `GroupResourceStatusException` for returning `null`.